src/Shipping/LocationRatesProcessor.php

```diff
@@ -56,21 +76,8 @@ public function process( array $location_rates ): array {
 	 * @return bool
 	 */
 	protected function should_rate_be_replaced( ShippingRate $new_rate, ShippingRate $existing_rate ): bool {
-		$replace = false;
-
-		if ( $new_rate instanceof ShippingRateFlat &&
-			 $existing_rate instanceof ShippingRateFlat &&
-			 $new_rate->get_rate() > $existing_rate->get_rate()
-		) {
-			$replace = true;
-		} elseif ( $new_rate instanceof ShippingRateFree &&
-				   $existing_rate instanceof ShippingRateFree &&
-				   (float) $new_rate->get_threshold() > (float) $existing_rate->get_threshold()
-		) {
-			$replace = true;
-		}
-
-		return $replace;
+		return $new_rate->get_rate() > $existing_rate->get_rate() ||
+			   (float) $new_rate->get_min_order_amount() > (float) $existing_rate->get_min_order_amount();
```

Comment:
This is intentional and specified in the requirements of this feature. The reasoning is that since we are grouping and estimating the rates it would be safer just to assume the most expensive rate for the group rather than applying the cheapest rate. For example, if a country-wide rate is $50 but a state-wide rate is $20, it would make more sense to apply the $50 rate to the whole country rather than the $20 rate.

Reply:
Thanks, @nima-karimi for the clarification, I got confused because [in this comment](https://github.com/woocommerce/google-listings-and-ads/issues/1395#issue-1190146411) we mentioned we would expect to return the cheapest shipping rate:

>Thailand (flat 25, free shipping 0)
Actual: {country: "TH", method: "free_shipping", currency: "USD", rate: 0, options: {}}
Expected (assuming **we are returning cheapest shipping rate**, hence discarding the "flat 25"): {country: "TH", method: "flat_rate", currency: "USD", rate: 0, options: {}}
🔴 Not match - should convert it to flat_rate.

I thought that we would follow the same idea, to suggest the cheapest and not the most expensive. 

On the other hand, I see the point that it is safer to show the most expensive although maybe it is not the best suggestion.

Reply:
Oh yeah, that's only the case if there is a free shipping method involved. Although I see the contradictions here. Given the same reasoning, one could argue that we should not return the free shipping but the most expensive flat-rate shipping for that group.

I think it would work best if we return the most expensive rate just to be on the safer side and don't surprise the user with a low rate on their Google profile. I can remove the condition that sets all rates to free if there is a free_shipping rate involved. @jorgemd24 @ecgan Do you think this would be preferable? Using the same Thailand example shared above, we would return the $20 rate instead of $0

Reply:
In my opinion, the issue is because it is difficult to make the balance between "safe" and a good suggestion only based on the rates that are set up in WC:

It sounds right that for example if Thailand has free shipping it should return a rate of $0 although, as you mentioned before, it is contradictory to the requirements of this feature. If we think from the "safer side", the free shipping never could apply if there is another flat rate > $0.

When there is more than one flat rate > $0, it is impossible to know which one is the right one so I suppose it is good to go with the most expensive, although this can go against the user's expectations. 

We need just to make sure that everyone understands that we are not returning the cheapest shipping rate, we return free shipping if exists otherwise we return the most expensive.

Reply:
Yes, it's a trade-off. And we're merely guessing the rates to make the onboarding smoother and also telling them to always check them in their Merchant Center settings after the onboarding is done. I think to be consistent it would be best to always return the most expensive rate regardless of whether they have a free-shipping rate or not. I've made some changes in https://github.com/woocommerce/google-listings-and-ads/pull/1422/commits/4f79a601e47dd46fbb9634a4b9297269214414af to always return the most expensive rate.